benchmark/fluid/fluid_benchmark.py

```diff
@@ -206,44 +215,62 @@ def train(avg_loss, infer_prog, optimizer, train_reader, test_reader, batch_acc,
             "fake data is not supported in single GPU test for now.")
 
     place = core.CPUPlace() if args.device == 'CPU' else core.CUDAPlace(0)
-    exe = fluid.Executor(place)
-    exe.run(startup_prog)
+
+    startup_exe = fluid.Executor(place)
+    startup_exe.run(startup_prog)
+    strategy = fluid.ExecutionStrategy()
+    strategy.num_threads = 1
+    strategy.allow_op_delay = False
+    train_exe = fluid.ParallelExecutor(
+        True, avg_loss.name, exec_strategy=strategy)
+
     feed_var_list = [
         var for var in train_prog.global_block().vars.itervalues()
         if var.is_data
     ]
     feeder = fluid.DataFeeder(feed_var_list, place)
 
     iters, num_samples, start_time = 0, 0, time.time()
-    for pass_id in range(args.pass_num):
-        train_losses = []
-        for batch_id, data in enumerate(train_reader()):
+    train_losses = []
+    if args.use_recordio:
```

Comment:
why the original training loop not able to use_recordio? Seems hacking the use_recordio into this place.